walk-in-interview/src/main/java/com/google/job/data/PaymentFrequency.java

```diff
@@ -0,0 +1,35 @@
+package com.google.job.data;
+
+/** Enumeration that represents the payment frequency. */
+public enum PaymentFrequency {
+    HOURLY("HOURLY"),
+    WEEKLY("WEEKLY"),
+    MONTHLY("MONTHLY"),
+    YEARLY("YEARLY");
+
+    private final String frequencyId;
+
+    PaymentFrequency(String frequencyId) {
+        this.frequencyId = frequencyId;
+    }
+
+    /**
+     * Gets the id string.
+     *
+     * @return Id string.
+     */
+    public String getFrequencyId() {
+        return frequencyId;
+    }
+
+    /**
+     * Returns the payment frequency enum matching the provided id.
+     *
+     * @param id String id matching the enum value.
```

Comment:
incorrect: the id is the frequencyId not the enum value. I actually think you can omit the `@param` altogether here, it's not providing much information over the code itself or the line above "Returns ..." which already provides the same information. The throws is important though.

Reply:
Fixed